### PR TITLE
Update setuptools requirements

### LINKED ISSUE #242: google-auth and google-api-core fail to install during setup
**Affecting:** 
Server

**Version:** 
b70d220dd2d2fcf91011d0eb7f40f774f1b5bfe8

**Describe the bug**
Installation using pip requirements file fails on Ubuntu 19.04 due to incompatible version of setuptools.

**To Reproduce**
1. `sudo apt-get install python3-pip`
2. `cd natlas/natlas-server && python3 -m venv venv`
3. `source venv/bin/activate`
4. `pip install -r requirements.txt`

**Expected behavior**
The requirements file should handle installation of the necessary version of setuptools. For some reason google-auth and/or google-api-core don't appear to include the setuptools versions in their actual dependencies to be installed, instead only printing this warning and then immediately crashing.

**Logs**
```
google-auth 1.7.1 has requirement setuptools>=40.3.0, but you'll have setuptools 28.8.0 which is incompatible.
google-api-core 1.14.3 has requirement setuptools>=34.0.0, but you'll have setuptools 28.8.0 which is incompatible.
Installing collected packages: MarkupSafe, Mako, SQLAlchemy, six, python-dateutil, python-editor, alembic, blinker, certifi, Click, dnspython, urllib3, elasticsearch, idna, email-validator, Werkzeug, itsdangerous, Jinja2, Flask, Flask-Login, Flask-Mail, Flask-SQLAlchemy, Flask-Migrate, WTForms, Flask-WTF, gunicorn, mpmath, netaddr, setuptools, protobuf, googleapis-common-protos, pyasn1, pyasn1-modules, rsa, cachetools, google-auth, chardet, requests, pytz, google-api-core, opencensus-context, opencensus, opencensus-ext-flask, grpcio, opencensus-proto, opencensus-ext-ocagent, opencensus-ext-sqlalchemy, Pillow, PyJWT, python-dotenv, python-libnmap, semver, sentry-sdk, sympy
  Running setup.py install for Mako ... done
  Running setup.py install for SQLAlchemy ... done
  Running setup.py install for alembic ... done
  Running setup.py install for blinker ... done
  Running setup.py install for Flask-Login ... done
  Running setup.py install for Flask-Mail ... done
  Running setup.py install for mpmath ... done
  Found existing installation: setuptools 40.8.0
    Uninstalling setuptools-40.8.0:
      Successfully uninstalled setuptools-40.8.0
  Running setup.py install for googleapis-common-protos ... error
    Complete output from command /home/dade/natlas/natlas/natlas-server/venv2/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-cg20rf8h/googleapis-common-protos/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-i3xsgnbh/install-record.txt --single-version-externally-managed --compile --install-headers /home/dade/natlas/natlas/natlas-server/venv2/include/site/python3.7/googleapis-common-protos:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/setuptools/__init__.py", line 10, in <module>
        from setuptools.extern.six.moves import filter, filterfalse, map
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/setuptools/extern/__init__.py", line 1, in <module>
        from pkg_resources.extern import VendorImporter
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3017, in <module>
        @_call_aside
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3003, in _call_aside
        f(*args, **kwargs)
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3045, in _initialize_master_working_set
        dist.activate(replace=False)
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2577, in activate
        declare_namespace(pkg)
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2151, in declare_namespace
        _handle_ns(packageName, path_item)
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2091, in _handle_ns
        _rebuild_mod_path(path, packageName, module)
      File "/home/dade/natlas/natlas/natlas-server/venv2/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2120, in _rebuild_mod_path
        orig_path.sort(key=position_in_sys_path)
    AttributeError: '_NamespacePath' object has no attribute 'sort'
    
    ----------------------------------------
Command "/home/dade/natlas/natlas/natlas-server/venv2/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-cg20rf8h/googleapis-common-protos/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-i3xsgnbh/install-record.txt --single-version-externally-managed --compile --install-headers /home/dade/natlas/natlas/natlas-server/venv2/include/site/python3.7/googleapis-common-protos" failed with error code 1 in /tmp/pip-install-cg20rf8h/googleapis-common-protos/
```

**Additional context**
It's unclear if this is a problem with ubuntu specifically or if other distributions might also encounter this problem. It's also a wonder that this hasn't been encountered during the docker builds for the latest versions. I guess it hasn't because the python packages are built off the python3 base image and then copied into the ubuntu image.

### PR DESCRIPTION
**Pull request by 0xdade**
Fixes #242 by including `setuptools==40.8.0` as an explicit dependency in requirements.txt

### CODE REVIEW

**Comment by 0xdade** [comment]
Docker build failed because the python 3.6.0 base image ships with pip 9. Somehow the combination of pip 9 and setuptools 28.8.0 made the googleapis package work fine, but pip 18 and setuptools 28.8.0 freaks it out. I was able to get docker to build again by manually updating the pip version and the setuptools version before installing the requirements file but I'm going to try to fix this by putting pip the required pip version in the requirements file.

I suspect that this may also be a matter of the way dependencies are listed in the requirements file controlling installation order because googleapis warns that it needs a specific setuptools version but doesn't actually require that setuptools version, it just fails.

Will report back once I've figured out the best way forward.

**Comment by 0xdade** [comment]
Updated dockerfile and requirements to avoid including setuptools as a project dependency since it creates significant problems in different environments. @ajacques can you please test this and see if this creates any issues on your end, since you're the one who added the setuptools dependency in the first place?

File: natlas-server/Dockerfile [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
 COPY . /opt/natlas/natlas-server
 
 RUN python3 -m compileall .
 
 # Build final image
-FROM python:3.6.0-slim
-
-RUN export DEBIAN_FRONTEND=noninteractive \
-	&& apt-get update \
-	&& apt-get install --no-install-recommends -qy libsqlite3-0
```
Comment (ajacques) [inline natlas-server/Dockerfile]: We don't need sqlite anymore? Does `python:3.7.5-slim` come with it?

Comment (0xdade) [reply natlas-server/Dockerfile]: Yeah, that build step was just printing that it was already installed.

**Pull request merged**